lib/v2/finviz/news.js

```diff
@@ -3,43 +3,68 @@ const cheerio = require('cheerio');
 const timezone = require('@/utils/timezone');
 const { parseDate } = require('@/utils/parse-date');
 
+const categories = {
+    news: 0,
+    blogs: 1,
+};
+
 module.exports = async (ctx) => {
-    const category = ctx.params.category ?? 'news';
+    const { category = 'News' } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 200;
+
+    if (!categories.hasOwnProperty(category.toLowerCase())) {
+        throw `No category '${category}'.`;
+    }
 
     const rootUrl = 'https://finviz.com';
-    const currentUrl = `${rootUrl}/news.ashx`;
+    const currentUrl = new URL('news.ashx', rootUrl).href;
 
-    const response = await got({
-        method: 'get',
-        url: currentUrl,
-    });
+    const { data: response } = await got(currentUrl);
 
-    const $ = cheerio.load(response.data);
+    const $ = cheerio.load(response);
 
-    const items = $('.table-fixed')
-        .eq(category === 'blog' ? 1 : 0)
-        .find('.nn')
+    const items = $('table.table-fixed')
+        .eq(categories[category.toLowerCase()])
+        .find('tr.nn')
+        .slice(0, limit)
         .toArray()
         .map((item) => {
             item = $(item);
 
-            const a = item.find('.nn-tab-link');
-
-            return {
-                title: a.text(),
-                link: a.attr('href'),
-                pubDate: timezone(parseDate(item.find('.nn-date').text(), ['MMM-DD', 'HH:mmA']), -4),
-                description:
-                    item
-                        .find('td[data-boxover]')
-                        .attr('data-boxover')
-                        ?.match(/<td class='news_tooltip-tab'>([\s\S]*)<\/td>/)[1] ?? '',
-            };
+            const a = item.find('a.nn-tab-link');
+
+            return ctx.cache.tryGet(a.prop('href'), () => {
```

Comment:
Some of the `(a.prop('href')` are `undefined`

Reply:
Unable to reproduce locally, I think I'll remove support for caching.

Reply:
I guess it's caused by Ads which are only visible to RSSHub.


<details><summary>Feed screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/e1bf6cef-a2b6-4575-809c-c6df972354d6)
</details>

Which can be solved by:
```diff
            const descriptionMatches = a
                .parent()
                .prop('data-boxover')
-                .match(/<td class='news_tooltip-tab'>(.*?)<\/td>/);
+                ?.match(/<td class='news_tooltip-tab'>(.*?)<\/td>/);
            const authorMatches = item
                .find('use')
                .first()
                .prop('href')
-                .match(/#(.*?)-(light|dark)/);
+                ?.match(/#(.*?)-(light|dark)/);
```
and a filter like `.filter((item) => item.title)`

Reply:
thanks. 😄